### PR TITLE
Add autoload paths for app plugins.

### PR DESCRIPTION
**Pull request by ADmad**
This change avoids having to update the app's composer.json to add autoload paths for app plugins under the "plugins" folder.

I have been using a custom composer plugin which does the same for quite a while for my apps.

### CODE REVIEW

**Comment by ADmad** [comment]
After this we can update bake's plugin creating command to run `composer dump-autoload` instead of modifying the app's `composer.json`.

**Comment by ADmad** [comment]
One can still manually update composer.json and any existing namespace under psr-4 autoload will be left unchanged (as seen for `Foo\\` and `Foo\Test\\` in the test case).

File: src/Plugin.php [hunk 1]
```diff
@@ -42,9 +42,55 @@ public static function getSubscribedEvents()
...
+
+                $pluginNamespace = $folderName . '\\';
+                $pluginTestNamespace = $folderName . '\\Test\\';
+                $path = $pluginsPath . '/' . $folderName . '/';
+                if (!isset($autoload['psr-4'][$pluginNamespace])) {
+                    $autoload['psr-4'][$pluginNamespace] = $path . 'src';
+                }
+                if (!isset($devAutoload['psr-4'][$pluginTestNamespace])) {
+                    $devAutoload['psr-4'][$pluginTestNamespace] = $path . 'tests';
+                }
```
Comment (markstory) [inline src/Plugin.php]: Should we make sure these directories exist before generating autoloader paths for them? The paths *should* exist but userland application structures are unpredictable.

Comment (ADmad) [reply src/Plugin.php]: Done

**Comment by markstory** [comment]
This is great :clap: We should announce this on the bakery as it is a good quality of life improvement.

**Comment by ADmad** [comment]
This is targeting 2.x. Would have to be backported to 1.x for Cake 4 apps.

**Comment by LordSimal** [comment]
If we are going to backport it to `1.x` I will definitely make a separate video showcasing the differences between the old and the new system. But indeed this is an awesome feature 👍🏻

**Comment by LordSimal** [comment]
This should be merged before we release Cake5

**Pull request merged**